app/validators/post_category_validator.rb

```diff
@@ -0,0 +1,7 @@
+class PostCategoryValidator < ActiveModel::Validator
+  def validate(record)
+    if record.category.has_children?
+      record.errors[:category_id] << "子カテゴリを持つカテゴリ「#{record.category.full_name}」に記事を登録することはできません。"
```

Comment:
How about using i18n for error message?
Such as https://github.com/bm-sms/nursepress/pull/272/commits/6c2a818594f35deff5f2a1f5b676f443db5c6634.